### PR TITLE
internal/ethapi: add testcases for blobTx

### PR DESCRIPTION
**Pull request by jsvisa**

### CODE REVIEW

**Comment by jsvisa** [comment]
@fjl ptal, thanks

**Comment by fjl** [comment]
It's good that you're adding tests, but I'm not so happy to have all of these test vectors inline as strings. If we ever change the API again, a lot of hand-editing will need to happen. So I would like to ask you, if you have time, to restructure these tests using `testdata` files. It can be done as follows:

- In the test case struct, add a new string field with a name. This name can be used as the "test name", but also as the filename of a JSON file in `testdata`.
- During test execution, if a special environment variable is set (e.g. `GENERATE_TEST`), the output that you are comparing against should be written to the testdata file. Use `json.MarshalIndent` to write a readable version.
- If the environment variable is not set, the test should load the file and compare the text just like now.

**Comment by jsvisa** [comment]
> It's good that you're adding tests, but I'm not so happy to have all of these test vectors inline as strings. If we ever change the API again, a lot of hand-editing will need to happen. So I would like to ask you, if you have time, to restructure these tests using `testdata` files. It can be done as follows:
> 
> 
> 
> - In the test case struct, add a new string field with a name. This name can be used as the "test name", but also as the filename of a JSON file in `testdata`.
> 
> - During test execution, if a special environment variable is set (e.g. `GENERATE_TEST`), the output that you are comparing against should be written to the testdata file. Use `json.MarshalIndent` to write a readable version.
> 
> - If the environment variable is not set, the test should load the file and compare the text just like now.


Sounds good to me, I'll adjust them this weekend

**Comment by jsvisa** [comment]
@fjl, please take another look, thanks.

use the env variable `WRITE_TEST_FILES` from https://github.com/ethereum/go-ethereum/blob/40ceae91f7098b93ed2f389ca924cb990a68e918/rlp/rlpgen/gen_test.go#L68

File: internal/ethapi/api_test.go [hunk 1]
```diff
@@ -88,6 +90,24 @@ func TestTransaction_RoundTripRpcJSON(t *testing.T) {
...
+		tests  = allTransactionTypes(common.Address{0xde, 0xad}, config)
+	)
+	testTransactionMarshal(t, tests, config)
+}
+
+func TestTransactionBlobTx(t *testing.T) {
+	newUint64 := func(val uint64) *uint64 { return &val }
+	config := *params.TestChainConfig
+	config.ShanghaiTime = newUint64(0)
+	config.CancunTime = newUint64(0)
```
Comment (holiman) [inline internal/ethapi/api_test.go:105]:
```suggestion
	config := *params.TestChainConfig
	config.ShanghaiTime = new(uint64)
	config.CancunTime = new(uint64)
```
Isn't this sufficient?

Comment (jsvisa) [reply internal/ethapi/api_test.go]: thanks

**Pull request merged**